### DISCUSSION TITLE
Organization's rewards contract

### DISCUSSION THREAD

**Discussion by L-Zuluaga** [stellar/sorobanathon#48] [General]
Mitkoin (MTK) is kommit's internal cryptocurrency, we use it to reward people for doing great tasks for the organization, such as contributing to open source, sharing knowledge and mentoring other engineers.

Currently, payments are done by hand through Stellar transactions, so we wanted to write a Soroban contract that would allow us to automate these rewards based on the organization's specific rules.

The contract code (including the contract tests) is available in this **[GitHub repository](https://github.com/kommitters/soroban_smart_contracts)**.

We went further and created a **[Gitbook](https://kommitters.gitbook.io/soroban-examples/)** explaining the entire contract workflow through an example.

Enjoy it :D!!! 

Cheers,
Lorenzo,
kommit - www.kommit.co

**Comment by achallagundla** [comment]
Love the gitbook addition to this! Will get back to you soon on your reward!

**Comment by achallagundla** [comment]
Heya @L-Zuluaga  ,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1000 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.

**Reply by L-Zuluaga** [reply]
@achallagundla thanks for the information, I wrote you a message in discord since we already filled the form.
